tests/unit/test_listserv.py

```diff
@@ -0,0 +1,131 @@
+import os
+import tempfile
+from pathlib import Path
+from unittest import mock
+
+import pytest
+import yaml
+
+import bigbang
+from bigbang import listserv
+from bigbang.listserv import ListservArchive, ListservList, ListservMessage
+from config.config import CONFIG
+
+dir_temp = tempfile.gettempdir()
+project_directory = str(Path(os.path.abspath(__file__)).parent.parent.parent)
+file_temp_mbox = dir_temp + "/listserv.mbox"
+file_auth = project_directory + "/config/authentication.yaml"
+auth_key_mock = {"username": "bla", "password": "bla"}
+
+
+class TestListservList:
+    @pytest.fixture(name="mlist", scope="module")
```

Comment:
Is this the intended usage of pytest fixtures? My understanding is that the fixture decorator doesn't belong on a test, but on another method that is used as a setup for some subset of tests.

(This comes up because currently some of the tests fail in the methods that are fixtures, and those are marked as errors, rather than test failures.)

Reply:
Yes, I agree. I haven't used it correctly.
What do you think about braking it up in the following way?:
```
@pytest.fixture(name="arch", scope="session")
def get_archive(self):
    arch = ListservArchive.from_listserv_directory(
        name="3GPP",
        directorypath=project_directory + "/tests/data/3GPP/",
    )
    return arch

def test__from_listserv_directory(self, arch):
    assert arch.name == "3GPP"
    mlist_names = [mlist.name for mlist in arch.lists]
    assert "3GPP_TSG_SA_ITUT_AHG" in mlist_names
    assert "3GPP_TSG_SA_WG2_MTCE" in mlist_names
    ahg_index = mlist_names.index("3GPP_TSG_SA_ITUT_AHG")
    mtce_index = mlist_names.index("3GPP_TSG_SA_WG2_MTCE")
    assert len(arch.lists[ahg_index]) == 25
    assert len(arch.lists[mtce_index]) == 57
    assert (
        arch.lists[mtce_index].messages[0].subject 
        == "test email - please ignore"
    )
```
Then the archive is initialized and can be used by all tests within the session but all tests are no moved into a test function without any fixture.

Reply:
Yeah, I think that's right. 

I'm not sure if the fixture means that the get_archive will get run separately for each of the tests (as opposed to using a setup step that would only get run once), but in any case the logical separation will mean that failures will get reported as the appropriate test failing, as opposed to a setup error issue.

Reply:
According to the [documention](https://docs.pytest.org/en/stable/fixture.html#fixture-scopes), when setting `scope="session"` the `get_archive` is only run once for and the `arch` can be used through the whole session. Saves a lot of time.